src/parser/jobs/nin/Ninki.js

```diff
@@ -0,0 +1,102 @@
+import React, {Fragment} from 'react'
+//import {Icon, Message} from 'semantic-ui-react'
+
+//import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+// Constants
+const MAX_NINKI = 100
+const NINKI_SPENDERS = [
+	ACTIONS.HELLFROG_MEDIUM.id,
+	ACTIONS.BHAVACAKRA.id,
+	ACTIONS.TEN_CHI_JIN.id,
+]
+
+export default class Ninki extends Module {
+	static handle = 'ninki'
+	static dependencies = [
+		'suggestions',
+	]
+	
+	_ninki = 0
+	_wastedNinki = 0
+	_wasteBySource = {
+		mug: 0,
+		auto: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		//this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		
+		if (abilityId === ACTIONS.MUG.id) {
+			this._wasteBySource.mug += this._addNinki(30)
+		}
+
+		if (abilityId === ACTIONS.ATTACK.id) {
+			this._wasteBySource.auto += this._addNinki(6)
+		}
+
+		if (NINKI_SPENDERS.includes(abilityId)) {
+			this._ninki -= 80
+		}
+	}
+
+	_addNinki(amt) {
+		// Helper for adding Ninki to the running tally and calculating waste. Returns the amount wasted.
+		this._ninki += amt 
+		if (this._ninki > MAX_NINKI) {
+			let waste = this._ninki - MAX_NINKI
+			this._wastedNinki += waste
+			this._ninki = MAX_NINKI
+			return waste 
+		}
+
+		return 0 
+	}
+
+/* Not used for now, but perhaps in the future
```

Comment:
Any reason? Without this in place, logs with a death will likely report a lot of wastage as the build up Ninki after the death.

Reply:
There was some debate on this last night in Discord. I ended up pulling it for now since we already have an established "death is bad" module and we couldn't figure out a good way to include it, phrasing-wise. If you've got a suggestion for keeping this baked in without it getting too messy, I'm open to changes. Maybe I could override the Death module and mention it in there?

Reply:
I'm not concerned about the actual output in the suggestion, just that the logic of death resetting your ninki is implemented.

Reply:
..._Shit_, good call. I feex.